### PR TITLE
put stuff in place to support app usage survey

### PR DESCRIPTION
**Pull request by anzioka**
PR:
- Implements two endpoints:
1) getAppUsageSurveyData(studyId, participantId, date) -> This queries chronicle_usage_events table in redshift and returns entries that match the specified parameters

TODO later: Investigate how to further filter the results (probably only return records of a certain type e.g Foreground Service). The aim is to entries that are a near-perfect representation of apps that a user actually interacted with

2) submitAppUsageSurvey(studyId, participantId, responses) -> writes the result of an app usage survey to postgres table

### CODE REVIEW

File: src/main/kotlin/com/openlattice/chronicle/services/ScheduledTasksManager.kt [hunk 1]
```diff
@@ -17,8 +22,18 @@ private const val DEVICES_INFO_REFRESH_INTERVAL = (60 * 1000L) // 1 minute
...
-    private val logger = LoggerFactory.getLogger(ScheduledTasksManager::class.java)
+class ScheduledTasksManager(
+        private val storageResolver: StorageResolver
+) {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ScheduledTasksManager::class.java)
+
+        private val GET_SYSTEM_APPS_SQL = """
+            SELECT * FROM ${SYSTEM_APPS.name}
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/services/ScheduledTasksManager.kt:33]: It's preferable to list out columns, even if you grab the SYSTEM_APPS.columns.joinToString

File: src/main/kotlin/com/openlattice/chronicle/services/surveys/SurveysService.kt [hunk 2]
```diff
@@ -49,16 +93,91 @@ class SurveysService(
...
+            return result
+
+        } catch (ex: Exception) {
+            logger.error("unable to parse date string")
+            throw ex
+        }
+    }
+
+    private fun updateAppUsage( organizationId: UUID, studyId: UUID, participantId: String, data: Map<UUID, Set<String>>): Int {
+        val (_, hds) = storageResolver.resolveAndGetFlavor(studyId)
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/services/surveys/SurveysService.kt:153]: Unless you are specifically planning to use redshift for survey storage, use the resolve with required flavor version of the call.

File: src/main/kotlin/com/openlattice/chronicle/storage/ChroniclePostgresTables.kt [hunk 3]
```diff
@@ -180,6 +188,52 @@ class ChroniclePostgresTables {
             .primaryKey(USER_ID)
             .overwriteOnConflict()
 
+        @JvmField
+        val APPS_USAGE = PostgresTableDefinition("app_usage")
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/storage/ChroniclePostgresTables.kt:192]: I already implemented app usage and storage inside of redshift. Check out redshift tables class and make sure you aren't duplicating that code.

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/chronicle/storage/ChroniclePostgresTables.kt:192]: Is this for storing the results of an app usage survey? If so I would update table name to reflect that so it is less confusing.

File: src/main/kotlin/com/openlattice/chronicle/services/surveys/SurveysService.kt [hunk 4]
```diff
@@ -49,16 +93,91 @@ class SurveysService(
...
+            ) {
+                ResultSetAdapters.appUsage(it)
+
+            }.toList()
+
+            logger.info(
+                    "fetched {} app usage entries for date {} $ORG_STUDY_PARTICIPANT",
+                    result.size,
+                    requestedDate,
+                    organizationId,
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/services/surveys/SurveysService.kt:139]: A study can have multiple organization ids, it seems you are assuming only one.

**Review by geekbeast — CHANGES_REQUESTED** [review]
Rename the app_usage table to app_usage_surveys for clarity. Otherwise looks good, but I have some comments/clarifying questions.

File: src/main/kotlin/com/openlattice/chronicle/storage/ChroniclePostgresTables.kt (hunk 3 above)
Comment (anzioka) [reply src/main/kotlin/com/openlattice/chronicle/storage/ChroniclePostgresTables.kt]: Yes, storing results of survey

**Review by geekbeast — APPROVED** [review]

**Pull request merged**